flocker/node/agents/functional/test_gce.py

```diff
@@ -78,7 +81,18 @@ def create_instance(self, instance_name):
             instance_name,
             machine_type=u"f1-micro"
         )
-        self.addCleanup(lambda: instance.destroy())
+
+        def destroy_best_effort(inst):
+            try:
+                inst.destroy()
+            except HttpError as e:
+                if e.resp.status == 404:
+                    # The test must have already destroyed the instance.
+                    pass
+                else:
+                    raise e
+
+        self.addCleanup(lambda: destroy_best_effort(instance))
```

Comment:
:question: nit this could be `self.addCleanup(destroy_best_effort, instance)` I think